pkg/quota/image/imagestreamimport_evaluator.go

```diff
@@ -17,64 +18,73 @@ import (
+	imageapi.ResourceImageStreams,
+}
+
+type imageStreamImportEvaluator struct {
+	store *oscache.StoreToImageStreamLister
+}
 
 // NewImageStreamImportEvaluator computes resource usage for ImageStreamImport objects. This particular kind
 // is a virtual resource. It depends on ImageStream usage evaluator to compute image numbers before the
 // the admission can work.
 func NewImageStreamImportEvaluator(store *oscache.StoreToImageStreamLister) kquota.Evaluator {
-	computeResources := []kapi.ResourceName{
-		imageapi.ResourceImageStreams,
-	}
-
-	matchesScopeFunc := func(kapi.ResourceQuotaScope, runtime.Object) bool { return true }
-
-	return &generic.GenericEvaluator{
-		Name:                       imageStreamImportName,
-		InternalGroupKind:          imageapi.Kind("ImageStreamImport"),
-		InternalOperationResources: map[admission.Operation][]kapi.ResourceName{admission.Create: computeResources},
-		MatchedResourceNames:       computeResources,
-		MatchesScopeFunc:           matchesScopeFunc,
-		UsageFunc:                  makeImageStreamImportAdmissionUsageFunc(store),
-		ListFuncByNamespace: func(namespace string, options metainternal.ListOptions) ([]runtime.Object, error) {
-			return []runtime.Object{}, nil
-		},
-		ConstraintsFunc: imageStreamImportConstraintsFunc,
+	return &imageStreamImportEvaluator{
+		store: store,
 	}
 }
 
-// imageStreamImportConstraintsFunc checks that given object is an image stream import.
-func imageStreamImportConstraintsFunc(required []kapi.ResourceName, object runtime.Object) error {
+// Constraints checks that given object is an image stream import.
+func (i *imageStreamImportEvaluator) Constraints(required []kapi.ResourceName, object runtime.Object) error {
 	if _, ok := object.(*imageapi.ImageStreamImport); !ok {
 		return fmt.Errorf("unexpected input object %v", object)
 	}
 	return nil
 }
 
-// makeImageStreamImportAdmissionUsageFunc returns a function for computing a usage of an image stream import.
-func makeImageStreamImportAdmissionUsageFunc(store *oscache.StoreToImageStreamLister) generic.UsageFunc {
-	return func(object runtime.Object) kapi.ResourceList {
-		isi, ok := object.(*imageapi.ImageStreamImport)
-		if !ok {
-			return kapi.ResourceList{}
-		}
-
-		usage := map[kapi.ResourceName]resource.Quantity{
-			imageapi.ResourceImageStreams: *resource.NewQuantity(0, resource.DecimalSI),
-		}
-
-		if !isi.Spec.Import || (len(isi.Spec.Images) == 0 && isi.Spec.Repository == nil) {
-			return usage
-		}
-
-		is, err := store.ImageStreams(isi.Namespace).Get(isi.Name)
-		if err != nil && !kerrors.IsNotFound(err) {
-			utilruntime.HandleError(fmt.Errorf("failed to list image streams: %v", err))
-		}
-		if is == nil || kerrors.IsNotFound(err) {
-			usage[imageapi.ResourceImageStreams] = *resource.NewQuantity(1, resource.DecimalSI)
-		}
-
-		return usage
+func (i *imageStreamImportEvaluator) GroupKind() schema.GroupKind {
+	return imageapi.Kind("ImageStreamImport")
```

Comment:
Hmmm... this will be tricky, b/c we should be returning both the old and new Kind, @mfojtik wdyt?